### PR TITLE
Remove two BIP32 vectors

### PR DESCRIPTION
**Pull request by junderw**
Two of the BIP32 vectors imply that the 4 byte version possibility space is limited.

However, the BIP itself is unclear about whether anything besides the 4 versions xpub, xprv, tpub, tprv is valid or not. Parenthesis is not sufficient.

Since BIP32 doesn't prescribe anything in regards to derivation scheme (it only suggests one as an example), or script type (it makes no mention and keeps the topic only on deriving keypairs).

So prescribing the version field as "it MUST be one of these 4 values" seems odd.

In the 11 years since adoption, there have been plenty of uses of BIP32 HD keys with alternative versions, both within and outside of the Bitcoin mainnet and testnet networks. (zprv etc. is one such example)

tl;dr We should make it clear that any version is valid BIP32, but that 4 specific ones are prescribed a meaning of "pertaining to mainnet" and "pertaining to testnet".

Alternatively: We should specify that bip32 should only treat those 4 versions as valid, all other MUST hard fail.

### CODE REVIEW

**Comment by luke-jr** [comment]
@sipa
